App crashing when selecting some specific images

**Summary:**

App is crashing whenever I am selecting specific image from gallery 


**Add System logs:**

java.lang.RuntimeException: Unable to start activity ComponentInfo{fr.free.nrw.commons.beta/fr.free.nrw.commons.upload.ShareActivity}: java.lang.NullPointerException: Attempt to invoke virtual method 'void java.io.OutputStream.close()' on a null object reference
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2781)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2859)
        at android.app.ActivityThread.-wrap11(Unknown Source:0)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1592)
        at android.os.Handler.dispatchMessage(Handler.java:106)
        at android.os.Looper.loop(Looper.java:164)
        at android.app.ActivityThread.main(ActivityThread.java:6518)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807)
     Caused by: java.lang.NullPointerException: Attempt to invoke virtual method 'void java.io.OutputStream.close()' on a null object reference
        at fr.free.nrw.commons.utils.FileUtils.saveFileFromURI(FileUtils.java:48)
        at fr.free.nrw.commons.utils.ContributionUtils.saveFileBeingUploadedTemporarily(ContributionUtils.java:34)
        at fr.free.nrw.commons.upload.ShareActivity.receiveImageIntent(ShareActivity.java:358)
        at fr.free.nrw.commons.upload.ShareActivity.onCreate(ShareActivity.java:299)
        at android.app.Activity.performCreate(Activity.java:7013)
        at android.app.Activity.performCreate(Activity.java:7004)
        at android.app.Instrumentation.callActivityOnCreate(Instrumentation.java:1215)
        at android.app.ActivityThread.performLaunchActivity(ActivityThread.java:2734)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2859) 
        at android.app.ActivityThread.-wrap11(Unknown Source:0) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1592) 
        at android.os.Handler.dispatchMessage(Handler.java:106) 
        at android.os.Looper.loop(Looper.java:164) 
        at android.app.ActivityThread.main(ActivityThread.java:6518) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.RuntimeInit$MethodAndArgsCaller.run(RuntimeInit.java:438) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:807) 

**Device and Android version:** 

Asus Zenfone Max Pro M1  Android 8.1.0
 
 **Commons app version:** 

master and betaDebug/prodDebug

**Would you like to work on the issue?**

Preferably no

--- a/app/src/main/java/fr/free/nrw/commons/utils/ContributionUtils.java
+++ b/app/src/main/java/fr/free/nrw/commons/utils/ContributionUtils.java
@@ -112,9 +112,9 @@ public class ContributionUtils {
             out = new FileOutputStream(new File(destinationFilename));
 
             byte[] buf = new byte[1024];
-            int len;
-            while ((len=in.read(buf)) > 0) {
-                out.write(buf, 0, len);
+            int length;
+            while ((length = in.read(buf)) > 0) {
+                out.write(buf, 0, length);
             }
         } catch (FileNotFoundException e) {
             e.printStackTrace();
@@ -123,6 +123,11 @@ public class ContributionUtils {
         } finally {
             try {
                 if (out != null) out.close();
+            } catch (IOException e) {
+                e.printStackTrace();
+            }
+
+            try {
                 if (in != null) in.close();
             } catch (IOException e) {
                 e.printStackTrace();

